### ISSUE TITLE
Compiler does not warn on unused locals on joined assignments

### ISSUE THREAD

**Issue by ivanbasov** [dotnet/roslyn#23322]
**Code example**
```
class C
{
    int M()
    {
        int a = 0;
        int b = 0;
        a = b = 0;
        return b;
    }
}
```
**Expected**
Provides a diagnostic that 'a' is unused

**Actual**
No diagnostics

roslyn-analyzers uses compiler warnings as a source of unused locals diagnostics

**Comment by alrz** [comment]
write is considered as a "use". what you want is "value assigned is not used in any execution path" warning from R#. that just doesn't exist in C# compiler out-of-the-box.

**Comment by jcouv** [comment]
Linking to warning waves issue (https://github.com/dotnet/roslyn/issues/1580)
